storage/elasticsearch/client.go

```diff
@@ -636,72 +642,86 @@ func (c *ElasticSearchClient) Stop() {
 func (c *ElasticSearchClient) Started() bool {
 	return c.started.Load() == true
 }
-func newBulkIndexer(c *elastigo.Conn, maxConns int, retrySeconds int, bulkMaxDocs int, bulkMaxDelay int) *elastigo.BulkIndexer {
-	indexer := c.NewBulkIndexer(maxConns)
-	indexer.RetryForSeconds = retrySeconds
-	if bulkMaxDocs > 0 {
-		indexer.BulkMaxDocs = bulkMaxDocs
+
+func newBulkIndexer(c *elastigo.Conn, connCfg ConnConfig) *elastigo.BulkIndexer {
+	indexer := c.NewBulkIndexer(connCfg.MaxConns)
+	indexer.RetryForSeconds = connCfg.RetrySeconds
+	if connCfg.BulkMaxDocs > 0 {
+		indexer.BulkMaxDocs = connCfg.BulkMaxDocs
 
 		// set chan to 80% of max doc
-		if bulkMaxDocs > 100 {
-			indexer.BulkChannel = make(chan []byte, int(float64(bulkMaxDocs)*0.8))
+		if connCfg.BulkMaxDocs > 100 {
+			indexer.BulkChannel = make(chan []byte, int(float64(connCfg.BulkMaxDocs)*0.8))
 		}
 	}
+
 	// override the default error chan size
 	indexer.ErrorChannel = make(chan *elastigo.ErrorBuffer, 100)
 
-	if bulkMaxDelay > 0 {
-		indexer.BufferDelayMax = time.Duration(bulkMaxDelay) * time.Second
+	if connCfg.BulkMaxDelay > 0 {
+		indexer.BufferDelayMax = time.Duration(connCfg.BulkMaxDelay) * time.Second
 	}
 
 	return indexer
 }
 
-// NewElasticSearchClient creates a new ElasticSearch client
-func NewElasticSearchClient(url *url.URL, maxConns int, retrySeconds int, bulkMaxDocs int, bulkMaxDelay int) (*ElasticSearchClient, error) {
+func urlFromHost(host string) (*url.URL, error) {
+	urlStr := host
+	if !strings.HasPrefix(urlStr, "http://") && !strings.HasPrefix(urlStr, "https://") {
```

Comment:
So host can be either hostname without http/s or full url ...  can you add explanation in default .yml configuration file

Reply:
This issue is outside the scope of this PR as i did not change the format of the skydive.yml file.

Additionally omitting the http prefix is used elsewhere (although not consistently) in other places such as analyzers.

For now i rather this not be addressed

Reply:
K, you are correct that this needs attention also in other places, I submitting a new issue to track https://github.com/skydive-project/skydive/issues/964

Reply:
ok